c/src/ml-api-service-agent-client.c

```diff
@@ -395,31 +395,28 @@ ml_service_launch_pipeline (const char *name, ml_service_h * handle)
   }
   *handle = NULL;
 
-  mls = g_try_new0 (ml_service_s, 1);
+  mls = _ml_service_create_internal (ML_SERVICE_TYPE_SERVER_PIPELINE);
   if (mls == NULL) {
     _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
         "Failed to allocate memory for the service handle. Out of memory?");
   }
 
-  server = g_try_new0 (_ml_service_server_s, 1);
+  mls->priv = server = g_try_new0 (_ml_service_server_s, 1);
   if (server == NULL) {
-    g_free (mls);
+    _ml_service_destroy_internal (mls);
     _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
         "Failed to allocate memory for the service handle's private data. Out of memory?");
   }
 
   ret = ml_agent_pipeline_launch (name, &(server->id), &err);
   if (ret < 0) {
-    g_free (server);
-    g_free (mls);
-    _ml_error_report ("Failed to invoke the method launch_pipeline (%s).",
+    _ml_service_destroy_internal (mls);
+    _ml_error_report_return (ret,
+        "Failed to invoke the method launch_pipeline (%s).",
         (err ? err->message : "Unknown error"));
-    return ret;
   }
 
   server->service_name = g_strdup (name);
-  mls->type = ML_SERVICE_TYPE_SERVER_PIPELINE;
-  mls->priv = server;
   *handle = mls;
```

Comment:
How about ```mls->priv = server``` in here?

Reply:
It is same mls->priv = server = new0().. at line 404.